### PR TITLE
Pcm 1019 typescript migration

### PR DESCRIPTION
**Pull request by djhouseknecht**

### CODE REVIEW

**Comment by djhouseknecht** [comment]
I moved `constants/index.js` and `utils/index.js` into the `/src` directory. I wasn't sure if that would cause problems? I can remove from what gets published to npm if I need to. I just thought it was more clear to have those in the source directory.

File: package.json [hunk 1]
```diff
@@ -1,69 +1,33 @@
...
     "dist/webrtc-sessions.js",
     "dist/webrtc-sessions.js.map"
   ],
   "scripts": {
     "build": "webpack && webpack --env.production",
-    "unit-test": "NODE_ENV=test nyc --reporter=lcov --reporter=cobertura ava test/unit/*.js",
-    "test": "semistandard && npm run unit-test && npm run report",
+    "test": "NODE_ENV=test npm run lint && jest --runInBand",
+    "test:watch": "NODE_ENV=test jest --watch --collectCoverage=false --runInBand",
+    "lint": "npm run semistandard && npm run tslint",
```
Comment (ckbrewer33) [inline package.json]: Does tslint not cover the same things that semistandard does?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
+{
+  "include": [
+    "./src/**/*",
+    "./test/**/*"
+  ],
+  "exclude": [
+    "./test/test-pages/**/*"
```
Comment (ckbrewer33) [inline tsconfig.json]: You can probably exclude node_modules here too

File: package.json (hunk 1 above)
Comment (djhouseknecht) [reply package.json]: Semistandard checks .js files only iirc. I pulled in semistandard's .ts rules. See here: https://github.com/purecloudlabs/purecloud-streaming-client-webrtc-sessions/pull/18/files#diff-ace19bd0c04529e685320269e3c05de9R2

**Comment by djhouseknecht** [comment]
Closing because it is stale. We will eventually migrate away from webrtc-sessions and use the new version of stanza which has TS support

**Pull request closed without merging**